spk/syncthing/src/dsm-control.sh

```diff
@@ -3,68 +3,94 @@
 # Package
 PACKAGE="syncthing"
 DNAME="Syncthing"
+DINAME="Syncthing-Inotify"
 
 # Others
 INSTALL_DIR="/usr/local/${PACKAGE}"
 PATH="${INSTALL_DIR}/bin:${PATH}"
 USER="syncthing"
+CONFIG_DIR="${INSTALL_DIR}/var"
+# Syncthing
 SYNCTHING="${INSTALL_DIR}/bin/syncthing"
-CONFIG_DIR="${INSTALL_DIR}/var/"
-
+SYNCTHING_OPTIONS="-home=${CONFIG_DIR}"
+# Syncthing-Inotify
+SYNCTHING_INOTIFY="${INSTALL_DIR}/bin/syncthing-inotify"
+SYNCTHING_INOTIFY_OPTIONS="-home=${CONFIG_DIR}"
+# Include user configurable options
+. ${CONFIG_DIR}/syncthing-addoptions.cfg
 
 start_daemon ()
 {
-    start-stop-daemon -b -o -c ${USER} -S -u ${USER} -x env HOME=${CONFIG_DIR} ${SYNCTHING} -- --home ${CONFIG_DIR}
+    start-stop-daemon -b -o -c ${USER} -S -u ${USER} -x env HOME=${CONFIG_DIR} $1 -- $2
 }
 
 stop_daemon ()
 {
-    start-stop-daemon -o -c ${USER} -K -u ${USER} -x ${SYNCTHING}
-    wait_for_status 1 20 || start-stop-daemon -K -s 9 -q -x ${SYNCTHING}
+    start-stop-daemon -o -c ${USER} -K -u ${USER} -x $1
+    wait_for_status $1 1 20 || start-stop-daemon -K -s 9 -q -x $1
 }
 
 daemon_status ()
 {
-    start-stop-daemon -K -q -t -u ${USER}
+    start-stop-daemon -K -q -t -u ${USER} -x $1
     [ $? -eq 0 ] || return 1
 }
 
 wait_for_status ()
 {
-    counter=$2
+    counter=$3
     while [ ${counter} -gt 0 ]; do
-        daemon_status
-        [ $? -eq $1 ] && return
+        daemon_status $1
+        [ $? -eq $2 ] && return
         let counter=counter-1
         sleep 1
     done
     return 1
 }
 
-
 case $1 in
     start)
-        if daemon_status; then
+        if daemon_status ${SYNCTHING}; then
             echo ${DNAME} is already running
         else
             echo Starting ${DNAME} ...
-            start_daemon
+			start_daemon ${SYNCTHING} "${SYNCTHING_OPTIONS}"
         fi
+        if daemon_status ${SYNCTHING_INOTIFY}; then
+            echo ${DINAME} is already running
+        else
+            echo Starting ${DINAME} ...
+            start_daemon ${SYNCTHING_INOTIFY} "${SYNCTHING_INOTIFY_OPTIONS}"
+		fi
```

Comment:
First, I just wanted to run inotify on my NAS. As soon as it should use the same user account to read config files with '600' rights, its presence breaks your start-stop-status script here:
```
daemon_status ()
{
    start-stop-daemon -K -q -t -u ${USER}
    [ $? -eq 0 ] || return 1
}
```
So, I decided to dig a bit more...